pdata/internal/generated_immutable_slice.go

```diff
@@ -0,0 +1,162 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by "model/internal/cmd/pdatagen/main.go". DO NOT EDIT.
+// To regenerate this file run "go run model/internal/cmd/pdatagen/main.go".
+
+package internal
+
+// ImmutableByteSlice represents a []byte slice that cannot be mutated.
+type ImmutableByteSlice struct {
+	value []byte
+}
+
+// NewImmutableByteSliceMoveFrom creates a new ImmutableByteSlice by replacing an underlying value of the []byte slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableByteSliceCopyFrom if it needs to be preserved.
+func NewImmutableByteSliceMoveFrom(val *[]byte) ImmutableByteSlice {
```

Comment:
Don't feel the best about these names for constructors, if anyone has an idea, will be appreciated. An option is to keep `func NewImmutableByteSlice(val []byte) ImmutableByteSlice` or maybe `func NewImmutableByteSlice(values ...byte) ImmutableByteSlice` since I believe that will force always to create a new "slice" so it will be easier to use than the current "CopyFrom" current form?

Reply:
I like `func NewImmutableByteSlice(values ...byte) ImmutableByteSlice` form

Reply:
I think we can keep `func NewImmutableByteSlice(values []byte) ImmutableByteSlice` form. I tried the `...byte` construction and looks like it doesn't force copying.

Reply:
Let's remove the move constructor for the moment? What do you think?

> I tried the ...byte construction and looks like it doesn't force copying.

Can you share the experiment or what you did?

Reply:
I updated the constructor to the following form:

```
func NewImmutableByteSlice(val ...byte) ImmutableByteSlice {
	is := ImmutableByteSlice{}
	is.value = val
	return is
}
```

Then tried this:

```
	b := []byte{1,2,3}
	is := pcommon.NewImmutableByteSliceM(b...)
	fmt.Println(is.At(0))
	b[0] = 10
	fmt.Println(is.At(0))
```

The output was:
```
1
10

```